Sample-01/app/profile/page.jsx

```diff
@@ -2,47 +2,37 @@
 
 import React from 'react';
 import { Row, Col } from 'reactstrap';
-import { useUser, withPageAuthRequired } from '@auth0/nextjs-auth0/client';
-
-import Loading from '../../components/Loading';
-import ErrorMessage from '../../components/ErrorMessage';
+import useAuthGuard from '../../hooks/useAuthGuard';
 import Highlight from '../../components/Highlight';
 
-function Profile() {
-  const { user, isLoading } = useUser();
+export default function Profile() {
+  const { isAuthenticated, user, guardComponent } = useAuthGuard();
+
+  // If not authenticated or loading, show guard component
+  if (!isAuthenticated) return guardComponent;
```

Comment:
I think introducing useAuthGuard here causes unneccesary complications in the diff.